enhancements/network/ovn-kubernetes-ipsec.md

```diff
@@ -0,0 +1,298 @@
+    ovnKubernetesConfig:
+      mtu: 1400
+      genevePort: 6081
+      **enableIPsec: true/false**
+
+### User Stories
+
+#### Story 1
+
+As a cluster operator, I want to be able to enable IPsec at cluster installation
+time in the OVNKubernetes plugin in order to encrypt, authenticate and ensure
+the integrity of all pod-pod traffic.
+
+### Implementation Details/Notes/Constraints [optional]
+
+It is expected that the following changes will need to be made in order to
+enable this enhancement:
+
+- OVNKubernetes pod Dockerfiles will need to be updated in order to install
+  necessary components required by OVN/OVS to enable IPsec (e.g.
+  openvswitch-ipsec, libreswan, etc).
+- Modify Cluster Network Operator manifests to configure start up of packages
+  required by OVN/OVS in order to enable IPsec (e.g. ovs-monitor-ipsec, pluto,
+  etc) and determine how to containerize these components.
+- Ensure that all versions of all components are at the correct level required
+  by OVN/OVS. This may required back-porting of patches into RHEL. It has
+  already been identified that OVS and the Kernel will each require a patch to
+  enable this enhancement.
+- Ensure correct generation of the Certificate Authority (CA) certificate and
+  all node private keys and CA-signed certificates. It is envisioned that the
+  key generation mechanism (and keys) of OVN SSL will be reused.
+- Modify Cluster Network Operator manifests to ensure that all OVS/OVN
+  components are configured to use host private key, host certificate and CA
+  certificate.
+- Modify OVNKubernetes operator configuration object in order to allow
+  enablement of IPsec for the cluster.
+- Certificate rotation will be handled in the same as it is with the current
+  OVN TLS implementation.
+
+### Risks and Mitigations
+
+#### Upstreaming
+
+A number of changes will need to be made to upstream communities including the
+Linux kernel. As these may need time to execute, it is imperative that we
+identify them early and start the process.
+
+#### Performance
+
+Enablement of IPsec will cause degradation of overall networking throughput and
+latency due to the additional overhead of the IPsec protocol. Some early
+performance testing will be done in a non-production cluster in order to get
+an approximation of the expected performance degradation - These results will be
+posted at a later date. After implementation of the enhancement, it is expected
+that comprehensive benchmarking and profiling will take place. Based on the
+output of this activity, it may be decided to start some optimization activities
+in parallel to improve performance.
+
+#### Security Review
+
+See "Open Questions" below.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Will this need to go through some kind of Security Review?
+
+### Test Plan
+
+It will expected to add a number of Continuous Integration jobs for IPsec:
+
+- All Openshift e2e tests should be replicated in a cluster that has been
```

Comment:
```suggestion
- All OpenShift e2e tests should be replicated in a cluster that has been
```

Reply:
Thanks, I will update this on the next revision